src/Document.php

```diff
@@ -87,4 +92,15 @@ public function __construct($data = [], $options = [])
             $this->clean();
         }
     }
+
+    /**
+     * Returns the Elastica\Result object that can be used for getting
+     * extra information about the document, such as the score and highlights.
```

Comment:
Wi would have to copy all methods from the Result object, though. My main concern is that there isn't always a Result (for example when you create the document yourself), what should all those methods return?